gammapy/cube/ring.py

```diff
@@ -313,51 +291,81 @@ def kernel(self, image):
         ring.normalize("peak")
         return ring
 
-    def run(self, images):
-        """Run ring background algorithm.
-
-        Required Maps: {required}
+    def make_maps_off(self, dataset, exclusion):
+        """Make off maps
 
         Parameters
         ----------
-        images : dict of `~gammapy.maps.WcsNDMap`
-            Input sky maps.
+        dataset : `~gammapy.cube.fit.MapDataset`
+            Input map dataset.
+        exclusion : `~gammapy.maps.WcsNDMap`
+            Exclusion mask for regions with known gamma-ray emission.
 
         Returns
         -------
-        result : dict of `~gammapy.maps.WcsNDMap`
-            Result sky maps
+        maps_off : dict of `~gammapy.maps.WcsNDMap`
+            Dictionary containing `counts_off` and `acceptance_off` maps.
         """
-        required = ["counts", "background", "exclusion"]
+        counts = dataset.counts
+        background = dataset.background_model.map
 
-        counts, background, exclusion = [images[_] for _ in required]
-
-        result = {}
+        maps_off = {}
         ring = self.kernel(counts)
 
         counts_excluded = counts * exclusion
-
-        result["off"] = counts_excluded.convolve(ring.array)
+        maps_off["counts_off"] = counts_excluded.convolve(ring.array)
 
         background_excluded = background * exclusion
+        maps_off["acceptance_off"] = background_excluded.convolve(ring.array)
 
-        result["exposure_off"] = background_excluded.convolve(ring.array)
+        # set pixels, where ring is too small to NaN
+        not_has_off_acceptance = maps_off["acceptance_off"].data <= 0
+        maps_off["acceptance_off"].data[not_has_off_acceptance] = np.nan
 
-        with np.errstate(divide="ignore", invalid="ignore"):
-            # set pixels, where ring is too small to NaN
-            not_has_off_exposure = result["exposure_off"].data <= 0
-            result["exposure_off"].data[not_has_off_exposure] = np.nan
+        not_has_acceptance = background.data <= 0
+        maps_off["counts_off"].data[not_has_acceptance] = 0
+        maps_off["acceptance_off"].data[not_has_acceptance] = 0
 
-            not_has_exposure = background.data <= 0
-            result["off"].data[not_has_exposure] = 0
-            result["exposure_off"].data[not_has_exposure] = 0
+        return maps_off
 
-            result["alpha"] = background / result["exposure_off"]
-            result["alpha"].data[not_has_exposure] = 0
+    def run(self, dataset, exclusion):
+        """Run ring background maker
 
-        result["background_ring"] = result["alpha"] * result["off"]
+        Parameters
+        ----------
+        dataset : `~gammapy.cube.fit.MapDataset`
+            Input map dataset.
+        exclusion : `~gammapy.maps.WcsNDMap`
```

Comment:
In the `ReflectedRegionsBackgroundMaker` the `exclusion_mask` is taken on `__init__`. I think for consistency it would be good to also move it to `RingBackgroundMaker.__init__()`. Alternatively we could also attach it to the dataset, but this we should discuss as separate issue.

Reply:
Done. In the new implementation, I had to add a `fill_by_coord()` to reproject internally the `exclusion_mask` to the cutout geom. With 4 runs, I checked that this increases the computation time of ~4%, which is not too bad I guess

Reply:
Yes, this is completely acceptable.